### PR TITLE
Add ODH deployment artifacts

### PR DESCRIPTION
**Pull request by anishasthana**
Many thanks for submitting your Pull Request :heart:!

This is a PR introducing the ability to deploy the trusty service via the ODH operator

cc @ruivieira

### CODE REVIEW

File: explainability-service/manifests/kfdef.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+apiVersion: kfdef.apps.kubeflow.org/v1
+kind: KfDef
+metadata:
+  name: odh-trusty
```
Comment (danielezonca) [inline explainability-service/manifests/kfdef.yaml]:
```suggestion
  name: odh-trustyai
```

File: explainability-service/manifests/kfdef.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    - kustomizeConfig:
+        repoRef:
+          name: odh-manifests
+          path: odh-common
+      name: odh-common
+    - kustomizeConfig:
+        repoRef:
+          name: manifests
+          path: explainability-service/manifests/opendatahub
+      name: trusty-ai
```
Comment (danielezonca) [inline explainability-service/manifests/kfdef.yaml]:
```suggestion
      name: trustyai
```

File: explainability-service/manifests/opendatahub/default/trusty-deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,118 @@
+---
```
Comment (danielezonca) [inline explainability-service/manifests/opendatahub/default/trusty-deployment.yaml]: Please rename the file as `trustyai-deployment.yaml`

File: explainability-service/manifests/opendatahub/default/trusty-deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+    app.openshift.io/vcs-url: <<unknown>>
```
Comment (danielezonca) [inline explainability-service/manifests/opendatahub/default/trusty-deployment.yaml]: Is this automatically replaced by some script? Should we populate it?

File: explainability-service/manifests/opendatahub/default/trusty-deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    app.quarkus.io/commit-id: ea73183a2c1b81bd9afe31005644d374b5bdeb34
```
Comment (danielezonca) [inline explainability-service/manifests/opendatahub/default/trusty-deployment.yaml]: Is this the actual commit id of the repo to be used? Should be update it for every release?

File: explainability-service/manifests/opendatahub/default/trusty-deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+    app.kubernetes.io/version: 999-SNAPSHOT
+  type: ClusterIP
+
+---
+apiVersion: image.openshift.io/v1
+kind: ImageStream
+metadata:
+  annotations:
+    app.openshift.io/vcs-url: <<unknown>>
+    app.quarkus.io/commit-id: ea73183a2c1b81bd9afe31005644d374b5bdeb34
```
Comment (danielezonca) [inline explainability-service/manifests/opendatahub/default/trusty-deployment.yaml]: Same comment here

File: explainability-service/manifests/opendatahub/default/trusty-deployment.yaml [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+spec:
+  dockerImageRepository: quay.io/ruimvieira/trustyai-service
+
+---
+apiVersion: apps.openshift.io/v1
+kind: DeploymentConfig
+metadata:
+  annotations:
+    app.openshift.io/vcs-url: <<unknown>>
+    app.quarkus.io/commit-id: ea73183a2c1b81bd9afe31005644d374b5bdeb34
```
Comment (danielezonca) [inline explainability-service/manifests/opendatahub/default/trusty-deployment.yaml]: And here

File: explainability-service/manifests/opendatahub/default/trusty-deployment.yaml [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+spec:
+  replicas: 1
+  selector:
+    app.kubernetes.io/name: trustyai-service
+    app.kubernetes.io/version: 999-SNAPSHOT
+  template:
+    metadata:
+      annotations:
+        app.openshift.io/vcs-url: <<unknown>>
+        app.quarkus.io/commit-id: ea73183a2c1b81bd9afe31005644d374b5bdeb34
```
Comment (danielezonca) [inline explainability-service/manifests/opendatahub/default/trusty-deployment.yaml]: And here

File: explainability-service/manifests/opendatahub/default/trusty-deployment.yaml [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+    prometheus.io/path: /q/metrics
+    prometheus.io/port: "8080"
+    prometheus.io/scheme: http
+  labels:
+    app.kubernetes.io/name: trustyai-service
+    app.kubernetes.io/version: 999-SNAPSHOT
+    app.openshift.io/runtime: quarkus
+  name: trustyai-service
+spec:
+  replicas: 1
```
Comment (danielezonca) [inline explainability-service/manifests/opendatahub/default/trusty-deployment.yaml]: @ruivieira 
We need to remember to test/fix concurrency and then update this to 3 or similar

**Comment by anishasthana** [comment]
image

I believe there's something wrong with the deploymentconfig triggers. It's kinda stuck in a deployment loop where we keep deploying more pods.

My vote would be to move away from the imagestream + deploymentconfig. Instead, we should just use a standard deployment with the image specified.

File: explainability-service/manifests/opendatahub/default/trusty-deployment.yaml [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+            - name: SERVICE_DATA_FORMAT
+              value: "CSV"
+            - name: MINIO_INPUT_FILENAME
+              value: "/inputs/income-biased-inputs.csv"
+            - name: MINIO_OUTPUT_FILENAME
+              value: "/inputs/income-biased-outputs.csv"
+            - name: SERVICE_METRICS_SCHEDULE
+              value: "5s"
+            - name: SERVICE_BATCH_SIZE
+              value: "5000"
```
Comment (danielezonca) [inline explainability-service/manifests/opendatahub/default/trusty-deployment.yaml]: @ruivieira @RobGeada @tteofili 
We need to remember to update these values as soon as we define/update the deployment mechanism

File: explainability-service/manifests/opendatahub/base/kustomization.yaml [hunk 11]
```diff
@@ -0,0 +1,7 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+commonLabels:
+  app: trusty-ai
+  app.kubernetes.io/part-of: trust-ai
```
Comment (danielezonca) [inline explainability-service/manifests/opendatahub/base/kustomization.yaml]:
```suggestion
  app: trustyai
  app.kubernetes.io/part-of: trustyai
```

File: explainability-service/manifests/opendatahub/default/kustomization.yaml [hunk 12]
```diff
@@ -0,0 +1,4 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+  - ./trusty-deployment.yaml
```
Comment (danielezonca) [inline explainability-service/manifests/opendatahub/default/kustomization.yaml]: To be renamed as 
```suggestion
  - ./trustyai-deployment.yaml
```
together with the file

File: explainability-service/manifests/opendatahub/default/trusty-deployment.yaml [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+      port: 80
+      targetPort: 8080
+  selector:
+    app.kubernetes.io/name: trustyai-service
+    app.kubernetes.io/version: 999-SNAPSHOT
+  type: ClusterIP
+
+---
+apiVersion: image.openshift.io/v1
+kind: ImageStream
```
Comment (danielezonca) [inline explainability-service/manifests/opendatahub/default/trusty-deployment.yaml]: Do we use `ImageStream` for ODH components? I was expecting to use a fixed (aka pre-build) image from registry 🤔

**Comment by anishasthana** [comment]
image
We're up to 91 😅

**Review by danielezonca — COMMENTED** [review]
Thanks a lot @anishasthana for the PR!
I have added some comments to better understand how it works and minor renaming stuff (I prefer to do them now because later is going to be harder)

**Comment by danielezonca** [comment]
> My vote would be to move away from the imagestream + deploymentconfig. Instead, we should just use a standard deployment with the image specified.

Ok this already answer to one of my comments :)

File: explainability-service/manifests/opendatahub/default/trusty-deployment.yaml (hunk 4 above)
Comment (anishasthana) [reply explainability-service/manifests/opendatahub/default/trusty-deployment.yaml]: Ill defer to @ruivieira  on this. I just copied his yamls :-)

File: explainability-service/manifests/opendatahub/default/trusty-deployment.yaml (hunk 5 above)
Comment (anishasthana) [reply explainability-service/manifests/opendatahub/default/trusty-deployment.yaml]: Ill defer to @ruivieira  on this. I just copied his yamls :-)

File: explainability-service/manifests/opendatahub/default/trusty-deployment.yaml (hunk 6 above)
Comment (anishasthana) [reply explainability-service/manifests/opendatahub/default/trusty-deployment.yaml]: Same as above

File: explainability-service/manifests/opendatahub/default/trusty-deployment.yaml (hunk 7 above)
Comment (anishasthana) [reply explainability-service/manifests/opendatahub/default/trusty-deployment.yaml]: Same as above

File: explainability-service/manifests/opendatahub/default/trusty-deployment.yaml (hunk 8 above)
Comment (anishasthana) [reply explainability-service/manifests/opendatahub/default/trusty-deployment.yaml]: Same as above

**Review by danielezonca — APPROVED** [review]
LGTM 👍

**Review by ruivieira — APPROVED** [review]

**Pull request merged**